satpy/readers/osisaf_l3_nc.py

```diff
@@ -0,0 +1,241 @@
+        x_size = self["/dimension/xc"]
+        y_size = self["/dimension/yc"]
+        p_lowerleft_lat = self["lat"].values[y_size - 1, 0]
+        p_lowerleft_lon = self["lon"].values[y_size - 1, 0]
+        p_upperright_lat = self["lat"].values[0, x_size - 1]
+        p_upperright_lon = self["lon"].values[0, x_size - 1]
+        area_extent = [p_lowerleft_lon, p_lowerleft_lat, p_upperright_lon, p_upperright_lat]
+        area_def = create_area_def(area_id="osisaf_polar_stereographic",
+                                   description="osisaf_polar_stereographic",
+                                   proj_id="osisaf_polar_stereographic",
+                                   projection=proj4str, width=x_size, height=y_size, area_extent=area_extent,
+                                   units="deg")
+        return area_def
+
+
+    def _get_finfo_grid(self):
+        """Get grid in case of filename info being used."""
+        if self.filename_info["grid"] == "ease":
+            self.area_def = self._get_ease_grid()
+            return self.area_def
+        elif self.filename_info["grid"] == "polstere" or self.filename_info["grid"] == "stere":
+            self.area_def = self._get_polar_stereographic_grid()
+            return self.area_def
+        else:
+            raise ValueError(f"Unknown grid type: {self.filename_info['grid']}")
+
+    def _get_ftype_grid(self):
+        """Get grid in case of filetype info being used."""
+        if self.filetype_info["file_type"] == "osi_radflux_grid":
+            self.area_def = self._get_geographic_grid()
+            return self.area_def
+        elif self.filetype_info["file_type"] == "osi_sst":
+            self.area_def = self._get_polar_stereographic_grid()
+            return self.area_def
+
+    def get_area_def(self, area_id):
+        """Override abstract baseclass method"""
+        if "grid" in self.filename_info:
+            return self._get_finfo_grid()
+        else:
+            return self._get_ftype_grid()
+
+
+    def _get_ds_attr(self, a_name):
+        """Get a dataset attribute and check it's valid."""
+        try:
+            return self[a_name]
+        except KeyError:
+            return None
+
+    def _get_ds_units(self, ds_info, var_path):
+        """Find the units of the datasets."""
+
+        file_units = ds_info.get("file_units")
+        if file_units is None:
+            file_units = self._get_ds_attr(var_path + "/attr/units")
+            if file_units is None:
+                file_units = 1
+        return file_units
+
+    def get_dataset(self, dataset_id, ds_info):
+        """Load a dataset."""
+        logger.debug(f"Reading {dataset_id['name']} from {self.filename}")
+        var_path = ds_info.get("file_key", f"{dataset_id['name']}")
+
+        shape = self[var_path + "/shape"]
+        data = self[var_path]
+        if shape[0] == 1:
+            # Remove the time dimension from dataset
+            data = data[0]
```

Comment:
Is the time dimension always size 1?

Reply:
Not all files have a time dimension, but if it is there it'll always be the first dim and always have length=1. Hence this check (copied from the GRHSST reader).